lib/edit-site-page.php

```diff
@@ -74,11 +79,53 @@ function gutenberg_edit_site_init( $hook ) {
 		$settings['fontSizes'] = $font_sizes;
 	}
 
-	// Get root template by trigerring `./template-loader.php`'s logic.
+	// Get all templates by triggering `./template-loader.php`'s logic.
+	$template_getters  = array(
+		'get_embed_template',
+		'get_404_template',
+		'get_search_template',
+		'get_home_template',
+		'get_privacy_policy_template',
+		'get_post_type_archive_template',
+		'get_taxonomy_template',
+		'get_attachment_template',
+		'get_single_template',
+		'get_page_template',
+		'get_singular_template',
+		'get_category_template',
+		'get_tag_template',
+		'get_author_template',
+		'get_date_template',
+		'get_archive_template',
+	);
+	$template_ids      = array();
+	$template_part_ids = array();
+	foreach ( $template_getters as $template_getter ) {
+		call_user_func( $template_getter );
+		apply_filters( 'template_include', null );
```

Comment:
Should we document this filter according to [WordPress documentation standards](https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#4-hooks-actions-and-filters)? It might make this logic easier to grok.

Reply:
Sure

Reply:
But we are not following that convention in the rest of the plugin, are we?

Reply:
Now that you mention it, I actually don't remember seeing it around.